src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingOperator.cs

```diff
@@ -132,13 +130,28 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
                 }
             }
 
-            // Optimize left ?? right to left.GetValueOrDefault() when left is T? and right is the default value of T
-            if (rewrittenLeft.Type.IsNullableType()
-                && RemoveIdentityConversions(rewrittenRight).IsDefaultValue()
-                && rewrittenRight.Type.Equals(rewrittenLeft.Type.GetNullableUnderlyingType(), TypeCompareKind.AllIgnoreOptions)
-                && TryGetNullableMethod(rewrittenLeft.Syntax, rewrittenLeft.Type, SpecialMember.System_Nullable_T_GetValueOrDefault, out MethodSymbol getValueOrDefault))
+            // Optimizations related to nullable value types
+            if (rewrittenLeft.Type.IsNullableType())
             {
-                return BoundCall.Synthesized(rewrittenLeft.Syntax, rewrittenLeft, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, getValueOrDefault);
+                var leftUnderlyingType = rewrittenLeft.Type.GetNullableUnderlyingType();
+                var unwrappedRight = RemoveIdentityConversions(rewrittenRight);
+                var rightIsOfUnderlyingTypeOfLeft = rewrittenRight.Type.Equals(leftUnderlyingType, TypeCompareKind.AllIgnoreOptions);
```

Comment:
>var rightIsOfUnderlyingTypeOfLeft = rewrittenRight.Type.Equals(leftUnderlyingType, TypeCompareKind.AllIgnoreOptions); [](http://example.com/codeflow?start=16&length=117)

It looks like this local and `leftUnderlyingType` are not necessary. We can simply restore the type comparison condition where it used to be and in its original form.